app/Main.hs

```diff
@@ -55,7 +59,12 @@ data Secret = Secret
 
 type EnvVar = (String, String)
 
-type ExecCtx = ([EnvVar], Options)
+data ExecCtx
+  = ExecCtx
+  { ctxEnv :: [EnvVar]
+  , ctxOpts :: Options
+  , ctxMan :: Manager
```

Comment:
I'm not a fan of the abbreviations here (or in general).

I'd go with `contextEnvironment`, `contextOptions`, `contextManager`.
Sure, it's a bit longer, but it's also unambiguous, which is more important imo.

E.g. `ctxMan` - is that the C transaction manual, perhaps?